### PR TITLE
Migrate to documenter jl, + doc upgrades

### PR DESCRIPTION
**Pull request by isentropic**
Moving to documnter jl as per zulip discussion

No deployment has been setup yet.

### CODE REVIEW

**Comment by isentropic** [comment]
@darsnack could you give me some feedback here please? Once we are good with docs I could proceed with functionality

**Comment by darsnack** [comment]
Sorry for the delay. Will take a look later today.

File: docs-documenter/src/buffering.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Buffering
+
+As mentioned in the section on [transformations](./tfminterface.md), you can implement [`apply!`](#) as an inplace version of [`apply`](#) to support buffered transformations. Usually, the result of a regular `apply` can be used as a buffer. You may write
```
Comment (darsnack) [inline docs-documenter/src/buffering.md]: References like this need to be mapped from Pollen to Documenter syntax. In general, you need to replace ``[`func`](#)`` with ``[`func`](@ref)``.

**Review by darsnack — COMMENTED** [review]
In general your mapping looks good. One global comment is to just replace the `docs` directory with Documenter instead of making a new Documenter specific directory. It looks like you have the syntax mapping figured out, but there are places you missed.

For deployment, see https://github.com/FluxML/ParameterSchedulers.jl/pull/58 as a reference for a similar transition. I'll take care of the admin side of Documenter deployment (keys, etc.).

File: docs-documenter/src/buffering.md (hunk 1 above)
Comment (isentropic) [reply docs-documenter/src/buffering.md]: yeah you are right, I just missed a few here and there. I'll proceed with fixing links and replacing the directory

**Comment by isentropic** [comment]
I did the changes you requested. I'm not good with yml files and github deployment, so I'm gonna leave up to you. Also currently the landing page link is broken when accessed from other repos. Not sure if it's the fault of other repos or this.

**Comment by darsnack** [comment]
@isentropic I have added the deployment stuff. The only stuff left in this PR is all related to editing the .md files for the documentation.

I've added flags to the `make.jl` script that will warn about issues when building the docs. In particular, Pollen would auto-populate an API reference page so that every ``[`func`](@ref)`` had an associated docstring. Documenter does not do this, so we need to manually insert the docstrings.

Can you pull the latest and re-run the doc build? You'll see the warnings telling you which functions/types you need to include a docstring for. If a page exists that's an appropriate home for the missing docstring, then add it there. For example, `DataAugmentation.ImageToTensor` can go on the `transformations.md` page. If an appropriate page does not exist, then create a new .md file for it. Use your best judgement and try to get to no more warnings. Then we can rearrange together after you push these changes.

**Comment by isentropic** [comment]
yeah, done

**Comment by isentropic** [comment]
@darsnack any updates on this?

**Review by CarloLucibello — APPROVED** [review]

**Comment by darsnack** [comment]
I should have posted an update. I was reorganizing things locally for this PR. I will make those changes on a new PR now.

Thanks for doing the lion's share of the work @isentropic!

**Pull request merged**